AirCasting/RootAppView.swift

```diff
@@ -11,24 +11,36 @@ import Resolver
 
 struct RootAppView: View {
 
-    @InjectedObject var userAuthenticationSession: UserAuthenticationSession
-    @InjectedObject var lifeTimeEventsProvider: LifeTimeEventsProvider
+    @InjectedObject private var userAuthenticationSession: UserAuthenticationSession
+    @InjectedObject private var lifeTimeEventsProvider: LifeTimeEventsProvider
+    @InjectedObject private var userState: UserState
+    @StateObject private var SignInPersistanceObserved = SignInPersistance.shared
     
     var body: some View {
         ZStack {
-            if userAuthenticationSession.isLoggedIn {
+            if userAuthenticationSession.isLoggedIn && userState.currentState != .loggingOut {
                 MainAppView()
-            } else if !userAuthenticationSession.isLoggedIn && lifeTimeEventsProvider.hasEverPassedOnBoarding {
-                NavigationView {
-                    CreateAccountView(completion: { self.lifeTimeEventsProvider.hasEverLoggedIn = true }).environmentObject(lifeTimeEventsProvider)
-                }
-            } else {
+                    .onAppear {
+                        SignInPersistanceObserved.clearDataWithCredentials()
+                    }
+            } else if !lifeTimeEventsProvider.hasEverPassedOnBoarding {
                 GetStarted(completion: {
                     self.lifeTimeEventsProvider.hasEverPassedOnBoarding = true
                 })
+            } else {
+                NavigationView {
+                    if SignInPersistanceObserved.signInActive {
+                        SignInView(completion: { self.lifeTimeEventsProvider.hasEverLoggedIn = true }).environmentObject(lifeTimeEventsProvider)
+                    } else {
+                        CreateAccountView(completion: { self.lifeTimeEventsProvider.hasEverLoggedIn = true }).environmentObject(lifeTimeEventsProvider)
+                    }
+                }
             }
         }
         .environment(\.managedObjectContext, Resolver.resolve(PersistenceController.self).viewContext) //TODO: Where is this used??
+        .onAppWentToBackground {
```

Comment:
Why did you add this here?

Reply:
So that we can prevent any credentials leak/steal thing. Without it, when you go to the background just for a while, you are able to go back and see those credentials as saved - it was not the case here. We should keep them on the screen (saved) only for the sake of allowing the user not to type them again when view refreshes - in any other case it needs to be deleted for safety